inc/API.php

```diff
@@ -42,12 +42,94 @@ public static function sync_user_if_enabled($user_id)
         $userdata['name'] = $user->first_name;
         $userdata['last_name'] = $user->last_name;
       }
-      self::peerboard_create_user($peerboard_options['auth_token'], $userdata);
+
+      $user = self::peerboard_create_user($peerboard_options['auth_token'], $userdata);
+
+      if (!$user) {
+        return;
+      }
+
       $count = intval(get_option('peerboard_users_count'));
       update_option('peerboard_users_count', $count + 1);
     }
   }
 
+  /**
+   * Check if API has error or not
+   *
+   * @param array or object $request
+   * @return void
+   */
+  public static function check_request_success($request, $function_args = [])
+  {
+    $success = true;
+
+    if (is_wp_error($request)) {
+      foreach ($request->errors as $notice => $message) {
+        peerboard_add_notice(sprintf('%s : %s', $notice, $message[0]), __FUNCTION__, 'error', $function_args);
+      }
+      $success = false;
+    }
+
+    if (is_array($request)) {
+      if ($request['response']['code'] >= 400) {
+        peerboard_add_notice($request['response']['message'], __FUNCTION__, 'error', $function_args);
+        $success = false;
+      }
+    }
+
+    return $success;
+  }
+
+  /**
+   * API Call
+   *
+   * @param [type] $slug
+   * @param [type] $token
+   * @param [type] $body
+   * @param string $type
+   * @return void
+   */
+  public static function peerboard_api_call($slug, $token = 0, $body, $type = 'GET', $api_url = '')
+  {
+    if (!empty($api_url)) {
+      $url = PEERBOARD_API_URL . $slug;
+    } else {
+      $url = PEERBOARD_API_BASE . $slug;
```

Comment:
It looks like there is a type and also I would use some url join function here instead of simple concatenation 
```suggestion
      $url = $api_url . $slug;
```